distributed/driver/driver.go

```diff
@@ -81,9 +84,12 @@ func (fcd *FlowDriver) RunFlowContext(parentCtx context.Context, fc *flow.Flow)
 	var wg, reportWg sync.WaitGroup
 	for _, taskGroup := range fcd.taskGroups {
 		wg.Add(1)
+		fcd.mu.RLock()
+		taskGroupsLen := len(fcd.taskGroups)
+		status := proto.Clone(fcd.GetTaskGroupStatus(taskGroup)).(*pb.FlowExecutionStatus_TaskGroup)
```

Comment:
move the logic into fcd.GetTaskGroupStatus() function? Maybe also adjust the function name to ReadTaskGroupStatus since the return result will be a clone.

Reply:
I'm currently simply solving existing data races, there's a lot of space for code improvements but I'd say they belong to a different PR